src/highdicom/sr/templates.py

```diff
@@ -1366,6 +2131,133 @@ def __init__(
+                'because it does not have Value Type CONTAINER.'
+            )
+        if get_coded_name(dataset) != codes.DCM.MeasurementGroup:
+            raise ValueError(
+                'Item #1 of sequence is not an appropropriate SR Content Item '
+                'because it does not have name "Measurement Group".'
+            )
+        instance = ContentSequence.from_sequence(sequence)
+        instance.__class__ = cls
+        return instance
+
+    @property
+    def method(self) -> Union[CodedConcept, None]:
+        """Union[highdicom.sr.coding.CodedConcept, None]: measurement method"""
+        root_item = self[0]
+        matches = find_content_items(
+            root_item,
+            name=codes.SCT.MeasurementMethod,
+            value_type=ValueTypeValues.CODE
+        )
+        if len(matches) > 0:
+            return matches[0].value
+        return None
+
+    @property
+    def tracking_identifier(self) -> Union[str, None]:
+        """Union[str, None]: tracking identifier"""
+        root_item = self[0]
+        matches = find_content_items(
+            root_item,
+            name=codes.DCM.TrackingIdentifier,
+            value_type=ValueTypeValues.TEXT
+        )
+        if len(matches) > 0:
+            return matches[0].value
+        return None
+
+    @property
+    def tracking_uid(self) -> Union[str, None]:
+        """Union[str, None]: tracking unique identifier"""
+        root_item = self[0]
+        matches = find_content_items(
+            root_item,
+            name=codes.DCM.TrackingUniqueIdentifier,
+            value_type=ValueTypeValues.UIDREF
+        )
+        if len(matches) > 0:
+            return matches[0].value
+        return None
+
+    @property
+    def finding_type(self) -> Union[CodedConcept, None]:
+        """Union[highdicom.sr.coding.CodedConcept, None]: finding"""
+        root_item = self[0]
+        matches = find_content_items(
+            root_item,
+            name=codes.DCM.Finding,
+            value_type=ValueTypeValues.CODE
+        )
+        if len(matches) > 0:
+            return matches[0].value
+        return None
+
+    @property
+    def finding_sites(self) -> List[FindingSite]:
+        """List[highdicom.sr.content.FindingSite]: finding sites"""
+        root_item = self[0]
+        matches = find_content_items(
+            root_item,
+            name=codes.SCT.FindingSite,
+            value_type=ValueTypeValues.CODE
+        )
+        if len(matches) > 0:
+            return [FindingSite.from_dataset(m) for m in matches]
+        return None
+
+    @property
+    def measurements(self) -> List[NumContentItem]:
+        """List[highdicom.sr.value_types.NumContentItem]: measurements"""
+        root_item = self[0]
+        return find_content_items(
+            root_item,
+            value_type=ValueTypeValues.NUM
+        )
+
+    @property
+    def qualitative_evaluations(self) -> List[CodeContentItem]:
```

Comment:
Return the new `QualitativeEvaluation` class?

Reply:
Not sure. It would then be logical to have `measurements` return `List[Measurement]`. However, a `Measurement` is a `Template` (and hence a `ContentSequence`), while `QualitativeEvaluation` is a `CodeContentItem`. That may be confusing. Thoughts?

Reply:
Oh boy... you've made me go and look more closely at the way measurement parsing has been implemented and opened up a can of worms.

I think my opinion here is that we should do what we can to mask the different between sequences and items from all but the most advanced users. It is only an important distinction to me when you are actually trying to piece the building blocks together the right way, which users should never be doing themselves. Consequently I think my suggestion would be that the `measurements` property or `get_measurements` method (whichever we go with) should probably return `Measurement` and furthermore `Measurement` should be given various properties to access its name, value, unit, tracking id, etc etc... Then the user really doesn't need to care whether what they get back is a sequence or an item.

I think the current situation in which when constructing a measurement the user provides, or may provide, all this extra information, but then on the parsing side only gets back the `NumContentItem` is quite confusing. There's currently no wway to access the `unit` at all, right? But that could be very important in many situations. Not much use getting the measurement without the unit, in my opinion.

I have done a similar thing to other sequences elsewhere, see `AlgorithmIdentifcationSequence` [here](https://github.com/MGHComputationalPathology/highdicom/blob/94d7c9d85ce5cf46b5230170ea8abe08782b0ff6/src/highdicom/content.py#L27),although that case is somewhat simpler.

To bring this back to the original point, I would also suggest returning `QualitativeEvaluation` here. I think generally having symmetry with encoding and decoding is going to make it easier for people to understand.

Sorry this complicates things when it felt like we were nearly there..., but I think this is a really important part to get right. What do you think?

Reply:
> I think my opinion here is that we should do what we can to mask the different between sequences and items from all but the most advanced users. It is only an important distinction to me when you are actually trying to piece the building blocks together the right way, which users should never be doing themselves. Consequently I think my suggestion would be that the measurements property or get_measurements method (whichever we go with) should probably return Measurement and furthermore Measurement should be given various properties to access its name, value, unit, tracking id, etc etc... Then the user really doesn't need to care whether what they get back is a sequence or an item.

Or we make an exception and implement the measurement content item in `highdicom.sr.content` derived from `NumContentItem` rather than implementing the TID 300 "Measurment" template in `highdicom.sr.templates`?

That would be more intuitive and also circumvent some problems related the inclusion of TID 300 in other templates. TID 300 is included in TID 1501 "Measurement and Qualitative Evaluation Group", but neither in TID 1410 "Planar ROI Measurements and Qualitative Evaluations" nor in TID 1411 "Volumetric ROI Measurements and Qualitative Evaluations".

This would break backwards compatibility, but I think it would be worth it. What do you think?

Reply:
I don't think I understand your proposal. How could you implement [TID 300](http://dicom.nema.org/medical/dicom/current/output/chtml/part16/chapter_A.html#sect_TID_300), with units, qualitifers, tracking IDs, etc etc, as a single NumContentItem?

Reply:
As discussed, I moved `QualitativeEvaluation` from `highdicom.sr.content` to `highdicom.sr.templates` and changed the base type from `CodeContentItem` to `Template` for consistency with `Measurement` (see 0b5ae3de258abb2c7c17ed3433fe607c9ece0d25). Now both `Measurement` and `QualitativeEvaluation` have a similar (awkward) interface.